src/librustdoc/html/render/print_item.rs

```diff
@@ -2267,6 +2271,53 @@ fn document_non_exhaustive<'a>(item: &'a clean::Item) -> impl fmt::Display + 'a
     })
 }
 
+fn document_default_header(
+    cx: &mut Context<'_>,
+    it: &clean::Item,
+    variants: &IndexVec<VariantIdx, clean::Item>,
+) -> String {
+    let tcx = cx.tcx();
+    let cache = cx.cache();
+    let v: Vec<&Impl> = cx.shared.all_impls_for_item(it, it.item_id.expect_def_id());
+
+    let default_fqp = vec![sym::core, kw::Default, sym::Default];
+    let mut label = "".to_string();
+    v.into_iter().for_each(|i| {
+        if let clean::ImplItem(ref impl_) = *i.impl_item.kind && let Some(trait_) = &impl_.trait_ {
+            let did = trait_.def_id();
+
+            let fqp = if let Some(&(ref fqp, _)) = cache.paths.get(&did) {
```

Comment:
You should be able to throw out everything related to `fqp` and `default_fqp`[^1] by making use of the fact that `Default` is a *diagnostic item*. Just do:

```rs
for i in v {
    if let clean::ImplItem(ref impl_) = *i.impl_item.kind
        && let Some(trait_) = &impl_.trait_
        && tcx.is_diagnostic_item(sym::Default, trait_.def_id())
    {
        let id = get_id_for_impl(tcx, i.impl_item.item_id);
        return chip(&format!("<a href=\"#{}\">Default</a>", id), "default");
    }
}
```

[^1]: `vec![sym::core, kw::Default, sym::Default]` is a bit fragile since `core` might not refer to libcore but to a crate defined by the user and libcore could've been renamed with `extern crate core as sth;`.

Reply:
done